components/ai_chat/resources/page/components/site_title/index.tsx

```diff
@@ -4,24 +4,87 @@
  * You can obtain one at https://mozilla.org/MPL/2.0/. */
 
 import * as React from 'react'
+import classnames from '$web-common/classnames'
+import { getLocale } from '$web-common/locale'
+import Icon from '@brave/leo/react/icon'
+import Button from '@brave/leo/react/button'
 
 import styles from './style.module.scss'
 import { SiteInfo } from '../../api/page_handler'
 
 interface SiteTitleProps {
   siteInfo?: SiteInfo
   favIconUrl?: string
+  onDisconnectClick?: () => void
+}
+
+type Timer = ReturnType<typeof setTimeout>
+
+const HOVER_TRANSITION_DELAY = 500
+
+interface ToolTipProps {
+  isVisible: boolean
+}
+
+function Tooltip(props: ToolTipProps) {
+  const tooltipStyles = classnames({
+    [styles.tooltip]: true,
+    [styles.tooltipVisible]: props.isVisible
+  })
+
+  return (
+    <div role="tooltip" id="page-content-warning-tooltip" className={tooltipStyles}>
+      {getLocale('pageContentWarning')}
+    </div>
+  )
 }
 
 function SiteTitle (props: SiteTitleProps) {
+  const [isTooltipVisible, setIsTooltipVisible] = React.useState(false)
+  const timerId = React.useRef<Timer | undefined>();
+
+  const showTooltipWithDelay = () => {
+    timerId.current = setTimeout(() => {
+      setIsTooltipVisible(true)
+    }, HOVER_TRANSITION_DELAY)
+  }
+
+  const hideTooltip = () => {
+    timerId.current && clearTimeout(timerId.current)
+    setIsTooltipVisible(false)
+  }
+
+  const handleOnKeyDown = (e: React.KeyboardEvent<HTMLDivElement>) => {
+    if (e.key === "Escape") {
+      setIsTooltipVisible(false)
+    }
+  }
+
    return (
     <div className={styles.box}>
-      <div className={styles.favIconBox}>
+      <Tooltip isVisible={isTooltipVisible} />
+      <div
+        aria-describedby='page-content-warning-tooltip'
+        tabIndex={0}
+        className={styles.favIconBox}
+        onMouseOver={showTooltipWithDelay}
+        onMouseOut={hideTooltip}
+        onTouchStart={() => setIsTooltipVisible(true)}
+        onTouchEnd={() => setIsTooltipVisible(false)}
+        onFocus={() => setIsTooltipVisible(true)}
+        onBlur={() => setIsTooltipVisible(false)}
+        onKeyDown={handleOnKeyDown}
+      >
         { props.favIconUrl && <img src={props.favIconUrl} /> }
       </div>
       <div className={styles.titleBox}>
         <p className={styles.title}>{props.siteInfo?.title}</p>
       </div>
+      <div className={styles.actions}>
+        <Button kind="plain" onClick={props.onDisconnectClick}>
+          <Icon name="link-broken" />
+        </Button>
+      </div>
```

Comment:
Since we've called this the "disconnect button", can we label it as so?

Reply:
Changed

Reply:
I'm referring to the fact there is no aria label, just an icon. Yet in the aria description for the tooltip the instruction is to click the "disconnect button". There is no mention of which button is the referred disconnect button.

Reply:
Added